File: 9-palindrome-number/9-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        num = str(x)
        left = 0
        right = len(num) - 1
        
        while left < right:
            if num[left] != num[right]:
                return False
            left += 1
            right -= 1
        
        return True
        
        
        

File: 9-palindrome-number/9-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        String numStr = Integer.toString(x);
        int start = 0;
        int end = numStr.length()-1;
        
        while (start < end) {
            if (numStr.charAt(start) != numStr.charAt(end)) {
                return false;
            }
            start++;
            end--;
        }
        return true;
    }
}